src/core/server/metrics/event_loop_delays/event_loop_delays_monitor.test.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/* eslint-disable dot-notation */
+jest.mock('perf_hooks');
+import { monitorEventLoopDelay } from 'perf_hooks';
+import { EventLoopDelaysMonitor } from './event_loop_delays_monitor';
+
+describe('EventLoopDelaysMonitor', () => {
+  beforeAll(() => {
+    jest.useFakeTimers('modern');
+    const mockNow = jest.getRealSystemTime();
+    jest.setSystemTime(mockNow);
+  });
+  afterEach(() => jest.clearAllMocks());
+  afterAll(() => jest.useRealTimers());
+
+  test('#constructor enables monitoring', () => {
+    const eventLoopDelaysMonitor = new EventLoopDelaysMonitor();
+    expect(monitorEventLoopDelay).toBeCalledTimes(1);
+    expect(eventLoopDelaysMonitor['loopMonitor'].enable).toBeCalledTimes(1);
+  });
+
+  test('#collect returns event loop delays histogram', () => {
+    const eventLoopDelaysMonitor = new EventLoopDelaysMonitor();
+    expect(eventLoopDelaysMonitor['loopMonitor'].disable).toBeCalledTimes(0);
+    expect(eventLoopDelaysMonitor['loopMonitor'].enable).toBeCalledTimes(1);
+    const histogramData = eventLoopDelaysMonitor.collect();
+    expect(eventLoopDelaysMonitor['loopMonitor'].disable).toBeCalledTimes(1);
+    expect(eventLoopDelaysMonitor['loopMonitor'].enable).toBeCalledTimes(2);
+    expect(eventLoopDelaysMonitor['loopMonitor'].percentile).toHaveBeenNthCalledWith(1, 50);
+    expect(eventLoopDelaysMonitor['loopMonitor'].percentile).toHaveBeenNthCalledWith(2, 75);
+    expect(eventLoopDelaysMonitor['loopMonitor'].percentile).toHaveBeenNthCalledWith(3, 95);
+    expect(eventLoopDelaysMonitor['loopMonitor'].percentile).toHaveBeenNthCalledWith(4, 99);
+
+    expect(Object.keys(histogramData)).toMatchInlineSnapshot(`
+      Array [
+        "min",
+        "max",
+        "mean",
+        "exceeds",
+        "stddev",
+        "fromTimestamp",
+        "lastUpdatedAt",
+        "percentiles",
```

Comment:
Not sure how far we want to go with unit testing (do we have IT test for this?), but maybe having the mocked `monitorEventLoopDelay` returning a mocked value to assert that we're properly using/mapping the correct values to our structure would make sense?

Reply:
Youre right the wiring of this function is not tested. I've added a unit test to ensure that.